_posts/2022-01-18-four-favorite-firmware-debug-tools.md

```diff
@@ -0,0 +1,128 @@
+
+## FTDI Cable 
+
+If there's an unsung hero of firmware development, it is the humble serial cable. Since the dawn of computing, the UART peripheral has been used as a debug port, a control bus for test systems, and a peripheral interface. It is a ubiquitous communication medium for embedded systems, and for the last two decades, no serial cable has been more ubiquitous than those containing chips by Future Technology Devices International, a.k.a FTDI. 
+
+### Feeling Talkative, Developer Inquires?
+The FTDI TTL-232R-3V3 debug cable is a very common thing to see plugged into a developer machine, or lying tangled on an electronics workbench. Originally a USB device company that made keyboards, joysticks, and mice, FTDI struck gold by creating one of the first cheap, widely available USB-UART interface ICs. It's an excellent example of a device that does one thing really, really well: it converts USB signals to UART, and vice versa. That's it! (Here's [a really neat interview Adafruit did with Fred Dart](https://blog.adafruit.com/2016/02/08/exclusive-interview-with-fred-dart-ceo-of-ftdi-ftdichip-ftdi-adafruit/), one the founder of FTDI. Fascinating look at some semiconductor history!) 
+
+One of the most useful things about the FTDI cable is its ability to provide target power in addition to comms. Each cable exposes a handy 5V output tapped off of the host computer's 5V USB output. It's limited at an output of 100mA, but this more than enough for many bare metal microcontroller applications. A simple on-target LDO stepping down to the proper IO voltage is enough to make this useful to you!
+
+### Now Don't Get It Twisted
+
+If you are responsible for designing PCBs in addition to firmware development, you probably know by now that it is a rite of passage to mix up the TX and RX signals on the FTDI cable, or to reverse the cable pinout. Reversing the pinout is generally a manageable headache. If there's no mechanical interference, you can typically flip the cable over and be on your merry way. Mixing up the TX and RX signals is a bit more of a headache. However, a set of zero ohm resistors on the TX/RX lines near the connector is generally a good practice. Odds are, you won't need them - but if you do, it's a heck of a lot easier than cutting and splicing in a bit of impromptu trace surgery. 
+
+## Conclusion
+
+At the request of the fine folks at Memfault, I've been forced to curtail the ode I wrote to my oscilloscope, though I’d be happy to sing you its praises in another post. 
```

Comment:
:grinning: feel free to cover your oscilloscope if you like! they are invaluable in certain cases for sure (also a good topic for an article by itself, "top uses for an oscilloscope in embedded engineering").

Reply:
This is getting long; I could probably write just as much about the scope. 

Another time - I promise!